pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/generators/protobuf/CodecParseMethodGenerator.java

```diff
@@ -114,55 +114,59 @@ static String generateParseInternalMethod(final String modelClassName, final Lis
-                                    // handle error cases here, so we do not do if statements in normal loop
-                                    // Validate the field number is valid (must be > 0)
-                                    if (field == 0) {
-                                        throw new IOException("Bad protobuf encoding. We read a field value of " + field);
-                                    }
-                                    // Validate the wire type is valid (must be >=0 && <= 5). Otherwise we cannot parse this.
-                                    // Note: it is always >= 0 at this point (see code above where it is defined).
-                                    if (wireType > 5) {
-                                        throw new IOException("Cannot understand wire_type of " + wireType);
-                                    }
-                                    // It may be that the parser subclass doesn't know about this field
-                                    if (f == null) {
-                                        if (strictMode) {
-                                            // Since we are parsing is strict mode, this is an exceptional condition.
-                                            throw new UnknownFieldException(field);
-                                        } else {
-                                            // We just need to read off the bytes for this field to skip it and move on to the next one.
-                                            skipField(input, ProtoConstants.get(wireType));
-                                        }
+                        // Given the wire type and the field type, parse the field
+                        switch (tag) {
+            $caseStatements
+                            default -> {
+                                // The wire type is the bottom 3 bits of the byte. Read that off
+                                final int wireType = tag & TAG_WRITE_TYPE_MASK;
+                                // handle error cases here, so we do not do if statements in normal loop
+                                // Validate the field number is valid (must be > 0)
+                                if (field == 0) {
+                                    throw new IOException("Bad protobuf encoding. We read a field value of " + field);
+                                }
+                                // Validate the wire type is valid (must be >=0 && <= 5). Otherwise we cannot parse this.
+                                // Note: it is always >= 0 at this point (see code above where it is defined).
+                                if (wireType > 5) {
+                                    throw new IOException("Cannot understand wire_type of " + wireType);
+                                }
+                                // It may be that the parser subclass doesn't know about this field
+                                if (f == null) {
```

Comment:
Not strictly related to this PR, but somewhat related... Current code is, basically:
```java
FieldDefinition f = getField(field);
switch (tag) {
    // number of cases for known fields
    default: {
        if (f == null) {
            // throw an exception, if strict mode, skip otherwise
        }
        throw new IOException()
    }
}
```

I wonder if `f == null` check should be right after `getField()`. Does it really make sense to parse a value, if the field is unknown?

Reply:
I guess this is because the current implementation skips unknown fields in the default: case of the switch when running in non-strict mode. That case can technically be moved before the switch right after we fetch the field definition. However, I don't see any benefits of doing so because the logic of the code will remain exactly the same, and the clarity (or absence thereof) won't change. Also, this will remove the default: branch from the switch which isn't really ideal because this may lead to hidden bugs. So I'd prefer to keep this part as it is now. Let me know what you think.

Reply:
My whole idea was to fail faster. If the field is unknown, it doesn't make sense to check the tag. The default handler will be preserved to throw an exception. For example, the field may be known, but its wire type is different for some reason. In this case, none of the switch cases will match, and the default handler will be used.

I agree this is not critical, but I'm still leaning towards moving the field check right before the switch statement.

Reply:
Fail fast is a good thing in general, I agree. Let me file a task under the PBJ Hardening epic to address this.

Reply:
I've just added it to https://github.com/hashgraph/pbj/issues/164

Reply:
@artemananiev : so, I closed https://github.com/hashgraph/pbj/issues/181 as WillNotFix. Please see the issue for more details.